### PR TITLE
Made all locations outputted by GithubDiscoveryProcessor optional

### LINKED ISSUE #4730: github-discovery should only output repositories with relevant files
Currently the way `github-discovery` is implemented is that it outputs all repositories that match a certain pattern, and leave it to a lower level to find the relevant catalog file within that repo. This causes a weird behavior if for example the following pattern is given:
```yaml
locations:
    - type: github-discovery
      target: https://github.com/backstage/*mkdocs/blob/master/catalog.yaml
```

And only some of the repos with `mkdocs` in their name contain a `catalog.yaml` file. All repos will be outputted, which will cause an error to be thrown for those repos where `catalog.yaml` file doesn't exist.

## Expected Behavior

Only relevant repos matching the **entire** filepattern should be outputted.

## Current Behavior

Errors are thrown when reading repositories that partially match the file pattern.

## Your Environment

- NodeJS Version (v12):
- Operating System and Version (e.g. Ubuntu 14.04):
- Browser Information:

### PR DESCRIPTION
**Pull request by nirga**
## Hey, I just made a Pull Request!

Fixes #4730.

This avoids errors when specifying locations as `https://github.com/backstage/*mkdocs/blob/master/catalog.yaml` where not all repositories actually contain a file named `catalog.yaml`, which is ok from the user's perspective - but currently throws an error.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)

### CODE REVIEW

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4737-xxbtn27mee4l3ksun0nyffzwfshepha6.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/603c961ffcb22d9645634b91

File: plugins/catalog-backend/src/ingestion/processors/GithubDiscoveryProcessor.ts [hunk 1]
```diff
@@ -94,7 +94,10 @@ export class GithubDiscoveryProcessor implements CatalogProcessor {
             type: 'url',
             target: `${repository.url}${catalogPath}`,
           },
-          false,
+          // Not all locations may actually exist, since the user defined them as a wildcard pattern.
+          // Thus, we omit them as optional and let the downstream processor find them while not outputting
```
Comment (freben) [inline plugins/catalog-backend/src/ingestion/processors/GithubDiscoveryProcessor.ts:98]:
```suggestion
          // Thus, we emit them as optional and let the downstream processor find them while not outputting
```

**Review by freben — APPROVED** [review]
A little typo, then ready to go!

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4737-xxbtn27mee4l3ksun0nyffzwfshepha6.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/603c961ffcb22d9645634b91

File: plugins/catalog-backend/src/ingestion/processors/GithubDiscoveryProcessor.ts (hunk 1 above)
Comment (nirga) [reply plugins/catalog-backend/src/ingestion/processors/GithubDiscoveryProcessor.ts]: oops! fixed

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4737-xxbtn27mee4l3ksun0nyffzwfshepha6.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/603c961ffcb22d9645634b91

**Pull request merged**